test/controllers/search_controller_test.rb

```diff
@@ -455,4 +455,391 @@ def source_filter_count(controller)
+    end
+
+    test 'geobox lat/long cannot exceed max limit' do
+      high_query = {
+        geobox: 'true',
+        geoboxMinLongitude: 180.1,
+        geoboxMinLatitude: 90.1,
+        geoboxMaxLongitude: 180.2,
+        geoboxMaxLatitude: 90.2
+      }.to_query
+      get "/results?#{high_query}"
+      assert_response :redirect
+      assert_equal flash[:error], "Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0."
+    end
+
+    test 'geobox min range limit' do
+      VCR.use_cassette('geobox min range limit',
+                        allow_playback_repeats: true,
+                        match_requests_on: %i[method uri body]) do
+        acceptable_query = {
+          geobox: 'true',
+          geoboxMinLongitude: -180.0,
+          geoboxMinLatitude: -90.0,
+          geoboxMaxLongitude: -179.9,
+          geoboxMaxLatitude: -89.9
+        }.to_query
+        get "/results?#{acceptable_query}"
+        assert_response :success
+        assert_nil flash[:error]
+      end
+    end
+
+    test 'geobox max range limit' do
+      VCR.use_cassette('geobox max range limit',
+                        allow_playback_repeats: true,
+                        match_requests_on: %i[method uri body]) do
+        acceptable_query = {
+          geobox: 'true',
+          geoboxMinLongitude: 179.9,
+          geoboxMinLatitude: 89.9,
+          geoboxMaxLongitude: 180.0,
+          geoboxMaxLatitude: 90.0
+        }.to_query
+        get "/results?#{acceptable_query}"
+        assert_response :success
+        assert_nil flash[:error]
+      end
+    end
+
+    test 'geobox min lat/long must be less than max lat/long' do
+      bad_values = {
+        geobox: 'true',
+        geoboxMinLongitude: 100.0,
+        geoboxMinLatitude: 80.0,
+        geoboxMaxLongitude: 29.6,
+        geoboxMaxLatitude: 56.9
+      }.to_query
+      get "/results?#{bad_values}"
+      assert_response :redirect
+      assert_equal flash[:error], "Bounding box minimum lat/long cannot exceed maximum lat/long."
+    end
+
+    test 'geodistance lat/long cannot fall below min limit' do
+      low_query = {
+        geodistance: 'true',
+        geodistanceLatitude: -90.1,
+        geodistanceLongitude: -180.1,
+        geodistanceDistance: '50mi' 
+      }.to_query
+      get "/results?#{low_query}"
+      assert_response :redirect
+      assert_equal flash[:error], "Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0."
+    end
+
+    test 'geodistance lat/long cannot exceed max limit' do
+      high_query = {
+        geodistance: 'true',
+        geodistanceLatitude: 90.1,
+        geodistanceLongitude: 180.1,
+        geodistanceDistance: '50mi' 
+      }.to_query
+      get "/results?#{high_query}"
+      assert_response :redirect
+      assert_equal flash[:error], "Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0."
+    end
+
+    test 'geodistance min range limit' do
```

Comment:
It took me a bit to understand what this test is doing - am I correct in understanding that this is testing that the lat/long values can go right up to the edge of the allowable envelope? (i.e. the validations don't kick in _at_ 90 or 180, but _after_ 90 or 190). I'm not sure what to do about this - I can't think of a better name, but that's what I find myself wanting - to make clear what is being tested here.

Reply:
Correct, the intent here is to test the minimum (and, in the other test, the maximum) allowable values. I'll retitle these tests to make that clearer.